packages/test-sync/build.gradle.kts

```diff
@@ -21,7 +21,32 @@ plugins {
     id("org.jetbrains.kotlin.multiplatform")
     id("com.android.library")
     kotlin("plugin.serialization") version Versions.kotlin
-    id("io.realm.kotlin")
+    // Test relies on the compiler plugin, but we cannot apply our full plugin from within the same
+    // gradle run, so we just apply the compiler plugin directly as a dependency below instead
+    // id("io.realm.kotlin")
+}
+
+
+// Test relies on the compiler plugin, but we cannot apply our full plugin from within the same
+// gradle run, so we just apply the compiler plugin directly
+dependencies {
```

Comment:
It is a bit magical and quite a lot of code, but since the end result is soo much better I think we can live with this until hopefully composite builds actually work in the IDE.